### ISSUE TITLE
Tutorial on fixing GAP problems

### ISSUE THREAD

**Issue by fkmoftakhar** [gap-system/gap#5137]
How to fix GAP errors? Is there a tutorial about this?

@fingolfin

**Comment by fingolfin** [comment]
I don't think we have such a thing yet, though perhaps someone else( @alex-konovalov @ThomasBreuer @frankluebeck @ChrisJefferson ...) is aware of something?

I agree it would be good to have something in this vein, i.e. "How to debug and fix errors in GAP", which at least gives some starting hints; like:
- how can I find out what code is executed by a a certain command (and point to e.g. `ApplicableMethod`)
- how can I find out in which file and line a function is defined
- ... or an operation ..
- ... or a method for an operation ...

and probably a bunch more

**Comment by fingolfin** [comment]
Also we should provide more powerful methods to help with this (and then point them out). E.g.

- it would be nice if `Edit(func)` would open the editor for the right file and line in which `func` is defined, if possible (see also issue #4749)
- the `Browse` package has some functions to help with this, too, which could be pointed out
- just discussed with @ChrisJefferson that we could modify `Display(func)` to print the file and line a function has been defined in (perhaps preface with `#` to ensure the output is still valid GAP code)
- likewise `Display(operation)` could print the location of all declarations of that operation / attribute / property
- ... and more

**Comment by frankluebeck** [comment]
I'm not aware of such a tutorial. But here are some more useful commands:
```
PageSource(Group); # in library
PageSource(SORT_LIST);  # in kernel
PageSource(ApplicableMethod(\in, [(1,2,3), SymmetricGroup(4)])); # method
```

**Comment by ThomasBreuer** [comment]
- There is [a chapter on debugging and profiling in the GAP Reference Manual](https://docs.gap-system.org/doc/ref/chap7.html).
- Concerning debugging individual GAP functions, the function `Debug` might be interesting. Currently it is not only undocumented but also not available in the GAP session by default; it becomes available by calling `ReadLib( "debug.g" )`.
- Concerning filename and line number of a GAP function, `PageSource` does show this information, but then a whole page of is shown, such that one can see the filename and line number only by scrolling up.

**Comment by fingolfin** [comment]
`LocationFunc` is useful to get the place a "classic" function was defined (it is not useful for operations/methods)
